src/ceph-volume/ceph_volume/devices/raw/activate.py

```diff
@@ -44,8 +45,49 @@ def activate_bluestore(meta, tmpfs, systemd):
     # correctly every time
     process.run(['ln', '-snf', meta['device'], os.path.join(osd_path, 'block')])
     system.chown(os.path.join(osd_path, 'block'))
-    system.chown(osd_path)
 
+    wal_device = os.getenv("CEPH_VOLUME_OSD_WAL_DEVICE")
+    if wal_device:
+        out, err, ret = process.call(['ceph-bluestore-tool', 'show-label', '--dev', wal_device])
+        if err:
+            terminal.error('ceph-bluestore-tool failed to run. %s'% err)
+            raise SystemExit(1)
+        if ret:
+            terminal.error('no label on %s'% wal_device)
+            raise SystemExit(1)
+        oj = json.loads(''.join(out))
+        if wal_device not in oj:
+            terminal.error('%s not in the output of ceph-bluestore-tool, buggy?'% wal_device)
+            raise SystemExit(1)
+        if oj[wal_device]['description'] == 'bluefs wal':
```

Comment:
i think this (and similar block below) needs to also verify the cluster and osd uuids match

Reply:
Only `osd_uuid` is shown by `ceph-bluestore-tool` for a wal/db device so I'll just check for `osd_uuid`.